packages/peregrine/lib/talons/CheckoutPage/OrderConfirmationPage/useCreateAccount.js

```diff
@@ -79,32 +77,25 @@ export const useCreateAccount = props => {
                         is_subscribed: !!formValues.subscribe
                     }
                 });
-
-                // Sign in and save the token
-                const response = await signIn({
+                const signInResponse = await signIn({
                     variables: {
                         email: formValues.customer.email,
                         password: formValues.password
                     }
                 });
-
-                const token =
-                    response && response.data.generateCustomerToken.token;
-
+                const token = signInResponse.data.generateCustomerToken.token;
                 await setToken(token);
-                await getUserDetails({ fetchUserDetails });
 
-                // Then remove the old guest cart and get the cart id from gql.
-                // TODO: This logic may be replacable with mergeCart in 2.3.4
+                // Clear guest cart cart from redux.
                 await removeCart();
 
-                await clearCartDataFromCache(apolloClient);
-                await clearCustomerDataFromCache(apolloClient);
-
+                // Create a new customer cart.
                 await createCart({
                     fetchCartId
                 });
 
+                // Ensure old stores are updated with any new data.
+                await getUserDetails({ fetchUserDetails });
```

Comment:
There's no `mergeCarts` operation here because the old "guest" cart will have just been ordered / wiped out, right?

Reply:
Yes - I had originally added it in the PR but realized there was no need to do that. If you create an account on the order confirmation page, you're doing so with zero items in your guest cart.